src/plugins/credential_management/public/components/create_credential_wizard/components/header/add_form.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSelect,
+  EuiLink,
+  EuiButton,
+  EuiFieldPassword,
+  EuiText,
+} from '@elastic/eui';
+import { Credential } from '../../../../../../data_source/public';
+export class AddForm extends Component<any, any> {
+  state: any = {
+    showErrors: false,
+    nameShowErrors: false,
+    errors: [],
+    toasts: [],
+    credentialName: '',
+    description: '',
+    credentialMaterialsType: Credential.CredentialMaterialsType.UsernamePasswordType,
+    userName: '',
+    password: '',
+    dual: true,
+  };
+
+  createCredential = async () => {
+    const { allList }: any = this.props;
+
+    const { credentialName, description, credentialMaterialsType, userName, password } = this.state;
+
+    const validationByField: any = [];
+    if (!credentialName) {
+      validationByField.push('Title should not be empty');
+    }
+    if (allList.find((item: any) => item.title === credentialName)) {
```

Comment:
what is the maximum number (rough idea) of credentials any user can have? 

Can we optimize this search ?